diff --git a/CTFd/api/v1/users.py b/CTFd/api/v1/users.py
index bd1d2ca..d3d0447 100644
--- a/CTFd/api/v1/users.py
+++ b/CTFd/api/v1/users.py
@@ -280,19 +280,34 @@ class UserPrivate(Resource):
     @users_namespace.doc(
         description="Endpoint to get the User object for the current user",
         responses={
-            200: ("Success", "UserDetailedSuccessResponse"),
             400: (
                 "An error occured processing the provided or stored data",
                 "APISimpleErrorResponse",
             ),
         },
     )
+    @users_namespace.response(
+        200,
+        "Success",
+        "UserDetailedSuccessResponse",
+        headers={
+            "X-CTFd-ID": "User ID",
+            "X-CTFd-Type": "User Type",
+        },
+    )
     def get(self):
         user = get_current_user()
         response = UserSchema("self").dump(user).data
         response["place"] = user.place
         response["score"] = user.score
-        return {"success": True, "data": response}
+        return (
+            {"success": True, "data": response},
+            200,
+            {
+                "X-CTFd-ID": user.id,
+                "X-CTFd-Type": user.type,
+            },
+        )
 
     @authed_only
     @users_namespace.doc(
